librz/core/cmd.c

```diff
@@ -4780,20 +4780,86 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(redirect_command) {
 	return res;
 }
 
+static bool do_help_manual(RzCmd *cmd, const char* cmdname) {
+	char *cmdname_help = NULL;
+	char *help_msg = NULL;
+	RzCmdParsedArgs *help_pra = NULL;
+	if (cmdname) {
+		cmdname_help = rz_str_newf("%s?", cmdname);
+		if (!cmdname_help) {
+			goto help_rec_err;
+		}
+		help_pra = rz_cmd_parsed_args_newcmd(cmdname_help);
+		if (!help_pra) {
+			goto help_rec_err;
+		}
+		help_msg = rz_cmd_get_help(cmd, help_pra, true);
+		if (!help_msg) {
+			goto help_rec_err;
+		}
+		rz_cons_printf("%s", help_msg);
+
+help_rec_err:
+		free (help_msg);
+		rz_cmd_parsed_args_free(help_pra);
+		free (cmdname_help);
+	}
+	return true;
+}
+
+static bool help_search_cmd_desc_entry(RzCmd *cmd, RzCmdDesc* cd, const char *name, void *user) {
+	rz_return_val_if_fail(cd, false);
+	if (cd->type != RZ_CMD_DESC_TYPE_ARGV &&
+		cd->type != RZ_CMD_DESC_TYPE_ARGV_MODES) {
+		return true;
+	}
+	PJ* pj = (PJ*) user;
+	if (pj) {
+		rz_cmd_get_help_json(cmd, cd, name, pj);
+	} else if (cd->type == RZ_CMD_DESC_TYPE_ARGV_MODES) {
+		do_help_manual(cmd, cd->name);
+	}
+	return true;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_help_search_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	bool is_interactive = rz_cons_is_interactive(); 
+	rz_cons_set_interactive(false); 
+	PJ* pj = NULL;
+	RzCmdStatus status = RZ_CMD_STATUS_OK;
+
+	RzCmdDesc *begin = argc == 2 ? rz_cmd_get_desc(core->rcmd, argv[1]) : NULL;
```

Comment:
avoid the ternary and just use a `if (argc == 2) ` where you set `begin = rz_cmd_get_desc` and you check `begin`.